django/joist/views.py

```diff
@@ -0,0 +1,30 @@
+from django.contrib.auth.models import User
+from django.http.response import HttpResponseBase
+from rest_framework import viewsets
+from rest_framework.decorators import api_view, parser_classes
+from rest_framework.parsers import JSONParser
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from .models import Blob
+from .serializers import BlobSerializer
+
+
+class BlobViewSet(viewsets.ModelViewSet):
+    queryset = Blob.objects.all()
+    serializer_class = BlobSerializer
+
+
+# @authentication_classes([TokenAuthentication])
+# @permission_classes([IsAuthenticated])
+@api_view(['POST'])
+@parser_classes([JSONParser])
+def save_blob(request: Request) -> HttpResponseBase:
+    creator = request.user if not request.user.is_anonymous else User.objects.first()
+    names = request.data.get('names', [])
+    out = []
+    for name in names:
```

Comment:
I think doing one finalize call per uploaded file allowed us to handle cases where some objects failed to upload to S3 without throwing away all the other successful objects.

Reply:
the main point was to separate the `finalize upload` from the actual saving of the model. Saving one file at a time is a valid approach and we have to think how to incorporate that in the library

Reply:
In the case of multiple files, I think we do want to finalize each of the files eagerly, as soon as they finish uploading to S3. We also want to save them independently, such that a single failure will not terminate the other uploads.

Whether we want to also upload them concurrently (possibly improving performance at the price of debuggability and simple linearizability).

All of this can be visited in the future, when we start considering what real scenarios for multiple files will look like.